lib/pbench/server/indexing_tarballs.py

```diff
@@ -0,0 +1,492 @@
+                                    "Indexing interrupted by SIGINT, continuing to next tarball"
+                                )
+                                continue
+                            finally:
+                                # Turn off the SIGINT handler when not indexing.
+                                signal.signal(signal.SIGINT, signal.SIG_IGN)
+                    except UnsupportedTarballFormat as e:
+                        idxctx.logger.warning("Unsupported tar ball format: {}", e)
+                        tb_res = 4
+                    except BadDate as e:
+                        idxctx.logger.warning("Bad Date: {!r}", e)
+                        tb_res = 5
+                    except FileNotFoundError as e:
+                        idxctx.logger.warning("No such file: {}", e)
+                        tb_res = 6
+                    except BadMDLogFormat as e:
+                        idxctx.logger.warning(
+                            "The metadata.log file is curdled in" " tar ball: {}", e
+                        )
+                        tb_res = 7
+                    except SosreportHostname as e:
+                        idxctx.logger.warning("Bad hostname in sosreport: {}", e)
+                        tb_res = 10
+                    except tarfile.TarError as e:
+                        idxctx.logger.error(
+                            "Can't unpack tar ball into {}: {}", ptb.extracted_root, e
+                        )
+                        tb_res = 11
+                    except SigTermException:
+                        idxctx.logger.exception(
+                            "Indexing interrupted by SIGTERM, terminating"
+                        )
+                        break
+                    except Exception as e:
+                        idxctx.logger.exception("Other indexing error: {}", e)
+                        tb_res = 12
+                    else:
+                        beg, end, successes, duplicates, failures, retries = es_res
+                        idxctx.logger.info(
+                            "done indexing (start ts: {}, end ts: {}, duration:"
+                            " {:.2f}s, successes: {:d}, duplicates: {:d},"
+                            " failures: {:d}, retries: {:d})",
+                            tstos(beg),
+                            tstos(end),
+                            end - beg,
+                            successes,
+                            duplicates,
+                            failures,
+                            retries,
+                        )
+                        tb_res = 1 if failures > 0 else 0
+                    try:
+                        ie_len = ie_filepath.stat().st_size
+                    except FileNotFoundError:
+                        # Above operation never made it to actual indexing, ignore.
+                        pass
+                    except SigTermException:
```

Comment:
Do we need to handle SigTermException here?

Reply:
just to be cautious

Reply:
@ndokos, see the comment on lines 323 and 324. If there is a better way to word that comment to make this requirement clear, let me know.